includes/server/class-llms-rest-memberships-controller.php

```diff
@@ -0,0 +1,590 @@
+			$page = get_post( $request['sales_page_page_id'] );
+			if ( $page && is_a( $page, 'WP_Post' ) ) {
+				$prepared_item['sales_page_content_page_id'] = $request['sales_page_page_id'];
+			} else {
+				$prepared_item['sales_page_content_page_id'] = 0;
+			}
+		}
+
+		// Sales page type.
+		if ( ! empty( $schema['properties']['sales_page_type'] ) && isset( $request['sales_page_type'] ) ) {
+			$prepared_item['sales_page_content_type'] = $request['sales_page_type'];
+		}
+
+		// Sales page URL.
+		if ( ! empty( $schema['properties']['sales_page_url'] ) && isset( $request['sales_page_url'] ) ) {
+			$prepared_item['sales_page_content_url'] = $request['sales_page_url'];
+		}
+
+		return $prepared_item;
+
+	}
+
+	/**
+	 * Prepare links for the request.
+	 *
+	 * @param LLMS_Membership $membership LLMS Membership.
+	 * @return array Links for the given object.
+	 */
+	protected function prepare_links( $membership ) {
+		$parent_links = parent::prepare_links( $membership );
+		$id           = $membership->get( 'id' );
+
+		$links = array();
+
+		// Enrollments.
+		$links['enrollments'] = array(
+			'href' => rest_url( sprintf( '/%s/%s/%d/%s', $this->namespace, $this->rest_base, $id, 'enrollments' ) ),
+		);
+
+		// Instructors.
+		$links['instructors'] = array(
+			'href' => add_query_arg(
+				'post',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'instructors' ) )
+			),
+		);
+
+		// Students.
+		$links['students'] = array(
+			'href' => add_query_arg(
+				'enrolled_in',
+				$id,
+				rest_url( sprintf( '%s/%s', 'llms/v1', 'students' ) )
+			),
+		);
+
+		return array_merge( $parent_links, $links );
+	}
+
+	/**
+	 * Prepare a single object output for response.
+	 *
+	 * @since [version]
+	 *
+	 * @param LLMS_Membership $membership Membership object.
+	 * @param WP_REST_Request $request    Full details about the request.
+	 * @return array
+	 */
+	protected function prepare_object_for_response( $membership, $request ) {
+		$data    = parent::prepare_object_for_response( $membership, $request );
+		$context = $request->get_param( 'context' );
+
+		// Auto enroll.
+		$data['auto_enroll'] = $membership->get_auto_enroll_courses();
+
+		// Catalog visibility.
+		$data['catalog_visibility'] = $membership->get_product()->get_catalog_visibility();
+
+		// Categories.
+		$data['categories'] = $membership->get_categories(
+			array(
+				'fields' => 'ids',
+			)
+		);
+
+		// Instructors.
+		$instructors         = $membership->get_instructors();
+		$instructors         = empty( $instructors ) ? array() : wp_list_pluck( $instructors, 'id' );
+		$data['instructors'] = $instructors;
+
+		// Restriction action.
+		$data['restriction_action'] = $membership->get( 'restriction_redirect_type' );
+		$data['restriction_action'] = $data['restriction_action'] ? $data['restriction_action'] : 'none';
+
+		// Restriction message.
+		$data['restriction_message'] = array(
```

Comment:
In my comment in `update_additional_object_fields()` I explain why I don't think the default restriction message shouldn't be handled here but on creation.